src/hooks/use-progress.tsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from 'react';
+
+interface UseProgressOptions {
+	isProcessing: boolean;
+	initialProgress?: number;
+	maxProgress?: number;
+	interval?: number;
+	step?: number;
+}
+
+export function useProgress( {
```

Comment:
ha!, I didn't remember we had `useProgressTimer`. Could we re-use it?

Reply:
The logic is quite similar, but not the same. But since we are not using the old version anymore, maybe we could just rename the new one to use the same name (and file name)? 🙂

Reply:
makes sense, let's leave it as it is. I like we are removing the useProgressTimer hook in favor of yours.

Reply:
Thank you, Antonio.